basil/HL/mercury.py

```diff
@@ -56,8 +75,42 @@ def get_channel(self, address=None):
         self._write_command("TS", address)
         return self.read()
 
-    def set_position(self, value, address=None):
+    def set_position(self, value, precision=100, address=None, wait=False):
         self._write_command("MA%d" % value, address)
+        if wait is True:
+            pos = self._wait(address)
+            if abs(pos - value) <= precision:
+                logger.info("At position {pos}, Target at {target}".format(pos=pos, target=value))
+            else:
+                logger.warning("Target not reached! Target: {target}, actual position: {pos}, precision: {pre}".format(target=value, pos=pos, pre=precision))
 
-    def move_relative(self, value, address=None):
+    def move_relative(self, value, precision=100, address=None, wait=False):
+        target = self.get_position(address=1) + value
         self._write_command("MR%d" % value, address)
+        if wait is True:
+            pos = self._wait(address)
+            if abs(pos - target) <= precision:
+                logger.info("At position {pos}, Target at {target}".format(pos=pos, target=target))
+            else:
+                logger.warning("Target not reached! Target: {target}, actual position: {pos}, precision: {pre}".format(target=target, pos=pos, pre=precision))
+
+    def abort(self, address=None):
+        self._write_command("AB", address)
+
+    def find_edge(self, n, address=None):
+        self._write_command("FE%d" % n, address)
+        pos = self._wait(address)
+        logger.info("Edge found at position: {pos}".format(pos=pos))
+
+    def _wait(self, address=None):  # waits until motor stops moving
+        logger.info("Moving! Starting position: {pos}".format(pos=self.get_position(address)))
+        done = False
+        while done is False:
+            a = self.get_position(address)
+            time.sleep(1)
+            b = self.get_position(address)
+            if a == b:
```

Comment:
Is this precise enough? I assume, some tolerance like one per mill might be good to account for numeric inaccuracies? Just wondering as this is typically the case for such that things.